features/request_specs/request_spec.feature

```diff
@@ -152,3 +152,23 @@ Feature: request spec
       """
     When I run `rspec spec`
     Then the example should pass
+
+  Scenario: testing subdomain constrained requests
+    Given a file named "spec/requests/widgets_spec.rb" with:
+      """ruby
+      require "rails_helper"
+
+      RSpec.describe "Widget management", :type => :request do
+        before { host! "api.example.com" }
```

Comment:
This test will pass without `host!`, too, won't it?

What do you think of drawing routes so that it fails without `host!`?
Something like
```ruby
Rails.application.routes.draw do
  resources :widgets, constraints: { subdomain: "api" }
end
```
should do.